Lib/importlib/_bootstrap_external.py

```diff
@@ -248,6 +248,21 @@ def _write_atomic(path, data, mode=0o666):
 #
 # Whenever MAGIC_NUMBER is changed, the ranges in the magic_values array
 # in PC/launcher.c must also be updated.
+#
+# In general, the MAGIC_NUMBER should not change for maintenance releases
+# although this may be required under exceptional circumstances. When
+# each release reaches candidate status, an entry in EXPECTED_MAGIC_NUMBERS
+# should be added for this release. This value is tested against the actual
+# MAGIC_NUMBER to ensure that if a change is required within a minor
+# release, the exception is first discussed with python-dev and relevant
+# community stakeholders such as OS distribution package maintainers
+# are properly informed of the change.
+
+EXPECTED_MAGIC_NUMBERS = {
+    (2, 7): 62211,
+    (3, 5): 3350,
+    (3, 6): 3379
+}
```

Comment:
I'd put this table in the test suite rather than in the code - there's no need to have it other than the bytecode stability test case, and if it hasn't been updated when it needs to be we'll get a test failure anyway.